engines/wintermute/ad/ad_scene.cpp

```diff
@@ -1960,16 +2004,38 @@ bool AdScene::scCallMethod(ScScript *script, ScStack *stack, ScStack *thisStack,
 	//////////////////////////////////////////////////////////////////////////
 	else if (strcmp(name, "GetLightName") == 0) {
 		stack->correctParams(1);
-		stack->pushNULL();
+
+		int index = stack->pop()->getInt();
+
+		if (_sceneGeometry) {
```

Comment:
This and the inner if could be merged safely as:
if (_sceneGeometry && index >= 0 && static_cast<uint>(index) < _sceneGeometry->_lights.size())

If you think that looks nicer.

Reply:
yes, eliminate double "stack->pushNULL();"

Reply:
Done